src/components/editMeasure/details/measureInformation/MeasureInformation.test.tsx

```diff
@@ -350,6 +350,48 @@ describe("MeasureInformation component", () => {
     expect(cmsId).toHaveProperty("readOnly", true);
   });
 
+  it("gracefully handles issue with generate cms id", async () => {
+    serviceApiMock = {
+      getAllEndorsers: jest.fn().mockResolvedValue(endorserList),
+      updateMeasure: jest.fn().mockResolvedValueOnce({ status: 200 }),
+      createCmsId: jest.fn().mockRejectedValueOnce({
+        status: 403,
+        response: { data: { message: "Failed to generate CMS ID." } },
+      }),
+    } as unknown as MeasureServiceApi;
+    useMeasureServiceApiMock.mockImplementation(() => serviceApiMock);
+
+    measure.measureSet = {
+      id: "id1",
+      measureSetId: "testMeasureId",
+      owner: "test.com",
+    };
+
+    render(<MeasureInformation setErrorMessage={setErrorMessage} />);
+    const result: HTMLElement = getByTestId("measure-information-form");
+    expect(result).toBeInTheDocument();
+
+    await act(async () => {
+      const cmsIdBtn = getByTestId(
+        "generate-cms-id-button"
+      ) as HTMLInputElement;
+      expect(cmsIdBtn).toBeEnabled();
+      act(() => {
+        fireEvent.click(cmsIdBtn);
```

Comment:
These are not a big deal, but I don't think we need act in both these scenarios.

Also userEvent is advised to be used instead of fireEvent unless we want to directly target the dom

Reply:
agreed, this was a copy/paste of another test and I just changed a few lines to fit the new scenario. But I have refactored the test to clean it up some